### PR TITLE
feat(CLDX-82): add checksum signing step

### PR DESCRIPTION
**Pull request by swickersh**
This commit adds a step to the sign-binaries task to generate the sha256 sums of all the signed binaries and collect them in a single file (sha256sum.txt).
It then transfers the file to a remote host to be signed with Red Hat's GPG key. The detached signatures are returned and released alongside binaries in subsequent tasks to the RH Developer Portal.

### CODE REVIEW

**Comment by swickersh** [comment]
@mmalina @johnbieren,

When @pkhander returns from PTO next week we'll get https://github.com/konflux-ci/release-service-catalog/pull/566 rebased. 
After that he and I will run the entire push-binaries pipeline several times. We plan to submit (hopefully) one last PR at that point with any fixes we encounter.  Prior to that, no products will be using this pipeline. 
Thanks!

**Review by johnbieren — COMMENTED** [review]
So do you want this merged in the meantime?

File: tasks/sign-binaries/tests/test-sign-binaries-checksum.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+              mkdir -p "$(workspaces.data.path)/content/linux"
+              cat > "$(workspaces.data.path)/content/linux/linux_binary.bin" << EOF
+              some data
+              EOF
+
+              # Create release.json with spec author
+              cat > "$(workspaces.data.path)/release.json" << EOF
+              {
+                "spec": {
+                  "author": "John Doe"
```
Comment (johnbieren) [inline tasks/sign-binaries/tests/test-sign-binaries-checksum.yaml]: author is not in the release spec so this is not really a good example https://github.com/konflux-ci/release-service/blob/main/api/v1alpha1/release_types.go

**Comment by swickersh** [comment]
> So do you want this merged in the meantime?

Yes, I don't see any reason to not go ahead and get it reviewed/merged.

File: tasks/sign-binaries/tests/test-sign-binaries.yaml [hunk 2]
```diff
@@ -9,6 +9,49 @@ spec:
...
+              mkdir -p "$(workspaces.data.path)/content/linux"
+              cat > "$(workspaces.data.path)/content/linux/linux_binary.bin" << EOF
+              some data
+              EOF
+
+              # Create release.json with spec author
+              cat > "$(workspaces.data.path)/release.json" << EOF
+              {
+                "spec": {
+                  "author": "John Doe"
```
Comment (johnbieren) [inline tasks/sign-binaries/tests/test-sign-binaries.yaml]: same author comment

**Review by johnbieren — DISMISSED** [review]
/ok-to-test

**Review by johnbieren — APPROVED** [review]

**Pull request merged**